### DISCUSSION TITLE
Implementing SwingTwist constraints for a human like character

### DISCUSSION THREAD

**Discussion by KyleN97** [jrouwe/JoltPhysics#1048] [Q&A]
Hey,

I've been trying to figure out the best way to setup the necessary SwingTwistConstraintSettings for my human ragdoll, It has a representation nearly identical to the human.tof example file. I've been looking through that file and it looks like these SwingTwistConstraintSettings where somewhat automatically calculated? is that right?

What would be the best way in figuring out the constraints settings, such as the following:
-  mTwistAxis1/2
- mPlaneAxis1/2
- mNormalHalfConeAngle
- mPlaneHalfConeAngle

As well as the necessary mSwingMotorSettings and mTwistMotorSettings?

I've implemented a quick system that allows me to manipulate the constraints on the fly but i've had no luck in really getting a good result. As you can see the SwingTwistConstraintSettings are all over the place at the moment.
![image](https://github.com/jrouwe/JoltPhysics/assets/19504651/155bf3e2-9f8b-40fa-a150-4153eb7c04d1)

**Comment by jrouwe** [comment]
> SwingTwistConstraintSettings where somewhat automatically calculated? is that right?

No they were placed in a tool by an artist and I just exported it to this format (the tool had functionality like aligning the rotation of a constraint with the joint and other things that help make the rig creation process easier).

There are several pieces of documentation that should help you determine what the settings should be:

* Motors: https://jrouwe.github.io/JoltPhysics/#constraint-motors
* Image of what the various axis mean: https://jrouwe.github.io/JoltPhysics/class_swing_twist_constraint_settings.html
* Creating a rig from code: https://github.com/jrouwe/JoltPhysics/blob/master/Samples/Tests/Rig/CreateRigTest.cpp
